src/uct/api/uct.h

```diff
@@ -414,6 +414,7 @@ enum {
 typedef enum {
     UCT_MD_MEM_TYPE_HOST = 0,      /**< Default system memory */
     UCT_MD_MEM_TYPE_CUDA,          /**< NVIDIA CUDA memory */
+    UCT_MD_MEM_TYPE_CUDA_MANAGED,  /**< NVIDIA CUDA managed (or unified) memory*/
```

Comment:
why do we need a new mem type? can't it be trated as HOST?
or maybe memory type should be flags instead of enum - e.g HOST, CUDA, HOST|CUDA (==managed)?

Reply:
@yosefe Yossi,  A similar question was asked earlier when the PR was opened. I answered it here: https://github.com/openucx/ucx/pull/2877/files/b49ef4aed8d880d2dfdf1e96113e62c45362c155#diff-a1f319ca66018bb8d03274d3b1757ec7R417

In short, we don't want RDMA on cuda-managed memory buffers unless it's some specific platform because it is incorrect to do so.  If HOST|CUDA == managed, then some UCT that claims support for both can incorrectly use RDMA on it, right?

Reply:
@Akshay-Venkatesh on which types of memory we want RDMA ?

Reply:
so we want RDMA on host, on CUDA, but not on managed?

Reply:
@yosefe That's correct.

Reply:
so probably i missed this, but how does this PR prevent RDMA on managed memory?

Reply:
managed_memory is not in list of any uct md cap.reg_mem_types capability flag.  with current protocols, it is fallack to AM rndv.

Reply:
Prior to this patch, cuda managed memory was treated like host memory so RDMA was a possibility. This PR removes that [translation](https://github.com/openucx/ucx/pull/2877/commits/825df7cedb700379c1f6596f49d6b40558583ee7#diff-8c5772ded80c8b4f61a934d4bcde22afL51). In addition, 
1. We introduce a new MD type for managed (needed for long term as well because we want to add a UCT that provides am_zcopy for managed memory)
2. We have none of the existing UCTs (some of which are RDMA capable) claim support for the MD

The result of 1. and 2. is that when memory is detected as managed, am_bcopy is used which forces packing and unpacking (using memcpy) for managed memory and hence no RDMA. Let me know if I've missed something.